romanisim/tests/test_l3.py

```diff
@@ -142,3 +143,364 @@ def test_inject_sources_into_mosaic():
+    # is the coordinate of the boresight, but that doesn't need to be on SCA 1.
+    # But at least they'll exercise some machinery if the ignore_distant_sources
+    # argument is high enough!
+    roman.n_pix = 100
+    exptime = 600
+
+    # Create metadata
+    meta = util.default_image_meta(filter_name='F158')
+    wcs.fill_in_parameters(meta, coord)
+    metadata = deepcopy(parameters.default_mosaic_parameters_dictionary)
+    filter_name = 'F158'
+    metadata['basic']['optical_element'] = filter_name
+    metadata['wcsinfo']['ra_ref'] = 270
+    metadata['wcsinfo']['dec_ref'] = 66
+
+    # Set up blank image
+    im = imdict['im'].copy()
+    im.array[:] = 0
+
+    # Set WCS
+    twcs = romanisim.wcs.get_mosaic_wcs(metadata, shape=(roman.n_pix, roman.n_pix))
+
+    # Create chromatic data in simulate_cps
+    im1 = im.copy()
+    im1, extras1 = l3.simulate_cps(im1, metadata, exptime, objlist=chromcat,
+                                   xpos=[50] * len(chromcat), ypos=[50] * len(chromcat),
+                                   bandpass=imdict['bandpass'], seed=rng_seed,
+                                   ignore_distant_sources=100)
+
+    # Create filter data in simulate_cps
+    im2 = im.copy()
+    im2, extras2 = l3.simulate_cps(im2, metadata, exptime, objlist=graycat,
+                                   xpos=[50] * len(chromcat), ypos=[50] * len(chromcat),
+                                   psf=imdict['impsfgray'], seed=rng_seed,
+                                   ignore_distant_sources=100)
+
+    # Ensure that the two simualtions are in line with each other
+    maxim = np.where(im1.array**2 > im2.array**2, im1.array, im2.array)
+    m = np.abs(im1.array**2 - im2.array**2) <= 20**2 * np.abs(maxim)
+    assert np.all(m)
+
+    # Create chromatic data in simulate
+    im3, extras3 = l3.simulate((roman.n_pix, roman.n_pix), twcs, exptime, filter_name, chromcat,
+                               bandpass=imdict['bandpass'], seed=rng_seed,
+                               cps_conv=1, unit_factor=(1 * u.MJy / u.sr),
+                               metadata=metadata, sky=0,
+                               ignore_distant_sources=100, effreadnoise=0,
+                               )
+
+    # Create filter data in simulate
+    im4, extras4 = l3.simulate((roman.n_pix, roman.n_pix), twcs, exptime, filter_name, graycat,
+                               psf=imdict['impsfgray'],
+                               cps_conv=1, unit_factor=(1 * u.MJy / u.sr), seed=rng_seed,
+                               metadata=metadata, sky=0,
+                               ignore_distant_sources=100,
+                               effreadnoise=0,
+                               )
+
+    # Ensure that the two simualtions are in line with each other
+    maxim = np.where(im3['data'].value**2 > im4['data'].value**2, im3['data'].value, im4['data'].value)
+    m = np.abs(im3['data'].value**2 - im4['data'].value**2) <= 20**2 * np.abs(maxim)
```

Comment:
Likewise.

Reply:
Removed this as the sim vs sim_cps test is the important one.